internal/pkg/expanders/storageBlob.go

```diff
@@ -0,0 +1,682 @@
+
+	containerID := currentItem.Metadata["ContainerID"]
+	containerName := e.getContainerName(containerID)
+	accountName := currentItem.Metadata["AccountName"]
+	accountKey := currentItem.Metadata["AccountKey"]
+	blobName := currentItem.Metadata["BlobName"]
+
+	blobEndpoint, err := e.getStorageBlobEndpoint(ctx, containerID)
+	if err != nil {
+		err = fmt.Errorf("Error getting blob endpoint: %s", err)
+		return ExpanderResult{
+			Err:               err,
+			SourceDescription: "StorageBlobExpander request",
+		}
+	}
+
+	// GetBlob docs: https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob
+	url := blobEndpoint + containerName + "/" + blobName
+	buf, err := e.doRequest(ctx, "GET", url, accountName, accountKey, "/"+accountName+"/"+containerName+"/"+blobName)
+
+	if err != nil {
+		return ExpanderResult{
+			Err:               fmt.Errorf("Error getting blob: %s", err),
+			SourceDescription: "StorageBlobExpander request",
+		}
+	}
+
+	result := string(buf)
+	return ExpanderResult{
+		Response:          ExpanderResponse{Response: result, ResponseType: ResponseXML},
+		SourceDescription: "StorageBlobExpander request",
+		Nodes:             []*TreeNode{},
+		IsPrimaryResponse: true,
+	}
+
+}
+
+func (e *StorageBlobExpander) deleteBlob(ctx context.Context, currentItem *TreeNode) (bool, error) {
+
+	containerID := currentItem.Metadata["ContainerID"]
+	containerName := e.getContainerName(containerID)
+	accountName := currentItem.Metadata["AccountName"]
+	accountKey := currentItem.Metadata["AccountKey"]
+	blobName := currentItem.Metadata["BlobName"]
+
+	blobEndpoint, err := e.getStorageBlobEndpoint(ctx, containerID)
+	if err != nil {
+		err = fmt.Errorf("Error getting blob endpoint: %s", err)
+		return false, err
+	}
+
+	// DeleteBlob docs: https://docs.microsoft.com/en-us/rest/api/storageservices/delete-blob
+	url := blobEndpoint + containerName + "/" + blobName
+	_, err = e.doRequest(ctx, "DELETE", url, accountName, accountKey, "/"+accountName+"/"+containerName+"/"+blobName)
+
+	if err != nil {
+		return false, fmt.Errorf("Error deleting blob: %s", err)
+	}
+
+	return true, nil
+
+}
+
+func (e *StorageBlobExpander) getAccountName(containerID string) string {
+	i := strings.Index(containerID, "/blobServices")
+	accountID := containerID[0:i]
+	i = strings.LastIndex(accountID, "/")
+	return accountID[i+1:]
+}
+func (e *StorageBlobExpander) getContainerName(containerID string) string {
+	i := strings.LastIndex(containerID, "/")
+	containerName := containerID[i+1:]
+	i = strings.Index(containerName, "?")
+	if i >= 0 {
+		containerName = containerName[:i] // strip query string
+	}
+	return containerName
+}
+
+func (e *StorageBlobExpander) getAccountKey(ctx context.Context, containerID string) (string, error) {
+
+	i := strings.Index(containerID, "/blobServices")
+	rootURL := containerID[0:i]
+	listKeysURL := rootURL + "/listKeys?api-version=2019-06-01"
+
+	data, err := e.armClient.DoRequest(ctx, "POST", listKeysURL)
```

Comment:
Would it make much difference to cache this result? We have the storage api with a ttl cache (or rough approximation of one) which could be used. Could key off the storage endpoint or accountid. 

Haven't dug into how regularly this call is made, if it's once up front then not worth it.

Reply:
Playing around with the code I didn't see any lag waiting on these calls so think this is me over-optimising.